diff -Naur linux-3.0.8_hisilicon/arch/arm/mach-godbox/core.c linux-3.0.8_hisilicon-PersianPrince/arch/arm/mach-godbox/core.c
--- linux-3.0.8_hisilicon/arch/arm/mach-godbox/core.c	2016-11-25 20:50:09.000000000 +0330
+++ linux-3.0.8_hisilicon-PersianPrince/arch/arm/mach-godbox/core.c	2021-04-04 18:44:13.768007458 +0430
@@ -36,12 +36,12 @@
 #include <linux/delay.h>
 #include <linux/android_pmem.h>
 #include <linux/mm.h>
-
+/*
 #ifdef CONFIG_PM
 extern unsigned long hi_pm_phybase;
 extern unsigned long hi_pm_ddrbase;
 #endif
-
+*/
 void __iomem *godbox_gic_cpu_base_addr;
 static unsigned long godbox_timer_reload;
 static unsigned long timer0_clk_hz;
@@ -495,6 +495,7 @@
 };
 static void __init godbox_init_early(void)
 {
+/*
 #if CONFIG_PM
 	extern unsigned long hi_pm_phybase;
 	extern unsigned long hi_pm_ddrbase;
@@ -506,6 +507,7 @@
 	}
 	hi_pm_phybase = __pa(hi_pm_ddrbase);
 #endif
+*/
 }
 
 MACHINE_START(GODBOX, "godbox")
diff -Naur linux-3.0.8_hisilicon/arch/arm/mach-godbox/hipm.c linux-3.0.8_hisilicon-PersianPrince/arch/arm/mach-godbox/hipm.c
--- linux-3.0.8_hisilicon/arch/arm/mach-godbox/hipm.c	2016-11-25 20:50:09.000000000 +0330
+++ linux-3.0.8_hisilicon-PersianPrince/arch/arm/mach-godbox/hipm.c	2021-04-04 18:45:02.080952297 +0430
@@ -16,13 +16,13 @@
 void __iomem *hi_sc_virtbase;
 void __iomem *hi_scu_virtbase;
 
-asmlinkage void hi_pm_sleep(void);
+//asmlinkage void hi_pm_sleep(void);
 
 static int hi_pm_enter(suspend_state_t state)
 {
 	hi_sc_virtbase = (void *)IO_ADDRESS(REG_BASE_SCTL);
 	hi_scu_virtbase = (void *)IO_ADDRESS(REG_BASE_A9_PERI);
-	hi_pm_sleep();
+//	hi_pm_sleep();
 	return 0;
 }
 
diff -Naur linux-3.0.8_hisilicon/arch/arm/mach-godbox/Makefile linux-3.0.8_hisilicon-PersianPrince/arch/arm/mach-godbox/Makefile
--- linux-3.0.8_hisilicon/arch/arm/mach-godbox/Makefile	2016-11-25 20:50:09.000000000 +0330
+++ linux-3.0.8_hisilicon-PersianPrince/arch/arm/mach-godbox/Makefile	2021-04-04 18:45:14.781200709 +0430
@@ -8,4 +8,4 @@
 
 obj-y					:= core.o clock.o cpu.o
 obj-$(CONFIG_HW_PERF_EVENTS)            += pmu.o
-obj-$(CONFIG_PM)			+= hi_pm_sleep.o hipm.o lowlevel_init.o
+obj-$(CONFIG_PM)			+= hipm.o lowlevel_init.o
