diff -Naur linux-3.0.8_hisilicon/Makefile linux-3.0.8_hisilicon-PersianPrince/Makefile
--- linux-3.0.8_hisilicon/Makefile	2016-11-25 20:50:09.000000000 +0330
+++ linux-3.0.8_hisilicon-PersianPrince/Makefile	2021-01-10 05:51:50.594841330 +0330
@@ -99,8 +99,8 @@
 # The O= assignment takes precedence over the KBUILD_OUTPUT environment
 # variable.
 
-ARCH := arm
-CROSS_COMPILE := arm-hisiv200-linux-
+#ARCH := arm
+#CROSS_COMPILE := arm-hisiv200-linux-
 
 # KBUILD_SRC is set on invocation of make in OBJ directory
 # KBUILD_SRC is not intended to be used by the regular user (for now)
